File: 36-valid-sudoku/valid-sudoku.java
class Solution {
    public boolean isValidSudoku(char[][] board) {
        for(int i = 0;i< board.length;i++){
            Set<Character> set = new HashSet<>();
            for(int j = 0;j<board[0].length;j++){               
                if(board[i][j] == '.'){
                    continue;
                }
                if(set.contains(board[i][j])){
                    return false;
                }
                set.add(board[i][j]);
            }
        }
        for(int j = 0;j< board[0].length;j++){
            Set<Character> set = new HashSet<>();
            for(int i = 0;i< board.length;i++){                 
                if(board[i][j] =='.'){
                    continue;
                }
                if(set.contains(board[i][j])){
                    return false;
                }
                set.add(board[i][j]);
            }
        }

        for(int i = 0;i < 9;i ++){
            Set<Character> set = new HashSet<>();
            for(int j = 0;j<9;j++){
                int m = i/3*3 + j/3;
                int n = i%3*3 + j%3;
                if(board[m][n] == '.') {
                    continue;
                }
                if(set.contains(board[m][n])) {
                    return false;
                }
                set.add(board[m][n]);
            }
        }

        return true;
    }
}

File: 36-valid-sudoku/valid-sudoku.js
/**
 * @param {character[][]} board
 * @return {boolean}
 */
var isValidSudoku = function(board) {
    
    
    for(let i = 0; i < 9; i++){

        const setRow = new Set();
        const setColumn = new Set();
        const setBox = new Set();

        for(let j = 0; j < 9; j++){

            let row = board[i][j];
            let column = board[j][i];
            let box = board[3 * (i % 3) + (j % 3)][3 * Math.floor(i / 3) + Math.floor(j / 3)];

            if(setRow.has(row) || setColumn.has(column) || setBox.has(box)){
                return false;
            }

            if(row !== "."){
                setRow.add(row)
            }

            if(column !== "."){
                setColumn.add(column)
            }

            if(box !== "."){
                setBox.add(box)
            }
        }   
    }

    return true;
};